tests/e2e/push_notifications/test_default_push_notification_support.py

```diff
@@ -0,0 +1,221 @@
+        except TimeoutError as e:
+            process.terminate()
+            raise e
+
+    yield url
+
+    process.terminate()
+    process.join()
+
+
+@pytest_asyncio.fixture(scope='function')
+async def http_client():
+    """An async client fixture for test functions."""
+    async with httpx.AsyncClient() as client:
+        yield client
+
+
+@pytest.mark.asyncio
+async def test_notification_triggering_with_in_message_config_e2e(
+    notifications_server: str, agent_server: str, http_client: httpx.AsyncClient
+):
+    """
+    Tests push notification triggering for in-message push notification config.
+    """
+    # Send a message with a push notification config.
+    response = await http_client.post(
+        f'{agent_server}/v1/message:send',
+        json={
+            'configuration': {
+                'pushNotification': {
+                    'id': 'n-1',
+                    'url': f'{notifications_server}/notifications',
+                    'token': uuid.uuid4().hex,
+                },
+            },
+            'request': {
+                'messageId': 'r-1',
+                'role': 'ROLE_USER',
+                'content': [{'text': 'Hello Agent!'}],
+            },
+        },
+    )
+    assert response.status_code == 200
+    task_id = response.json()['task']['id']
+    assert task_id is not None
+
+    # Retrive and check notifcations.
+    notifications = await wait_for_n_notifications(
+        http_client,
+        f'{notifications_server}/tasks/{task_id}/notifications',
+        n=2,
+    )
+    states = [notification['status']['state'] for notification in notifications]
+    assert 'completed' in states
+    assert 'submitted' in states
+
+
+@pytest.mark.asyncio
+async def test_notification_triggering_after_config_change_e2e(
+    notifications_server: str, agent_server: str, http_client: httpx.AsyncClient
+):
+    """
+    Tests notification triggering after setting the push notificaiton config in a seperate call.
+    """
+    # Send an initial message without the push notification config.
+    response = await http_client.post(
+        f'{agent_server}/v1/message:send',
+        json={
+            'request': {
+                'messageId': 'r-1',
+                'role': 'ROLE_USER',
+                'content': [{'text': 'How are you?'}],
+            },
+        },
+    )
+    assert response.status_code == 200
+    assert response.json()['task']['id'] is not None
+    task_id = response.json()['task']['id']
+
+    # Get the task to make sure that further input is required.
+    response = await http_client.get(f'{agent_server}/v1/tasks/{task_id}')
+    assert response.status_code == 200
+    assert response.json()['status']['state'] == 'TASK_STATE_INPUT_REQUIRED'
+
+    # Set a push notification config.
+    response = await http_client.post(
+        f'{agent_server}/v1/tasks/{task_id}/pushNotificationConfigs',
+        json={
+            'parent': f'tasks/{task_id}',
```

Comment:
Technically you don't specify this parameter in the body, it's taken from the path.